requirements.txt

```diff
@@ -1,12 +1,8 @@
-#docker minor version updates can include breaking changes. Auto update micro version only.
-# TODO: lucashuy
-# revisit the docker pin when docker fixes the below request issue
-docker~=4.2.0
+# docker minor version updates can include breaking changes. Auto update micro version only.
+docker~=7.1.0
```

Comment:
never paid attention to this before but curious why do we need docker sdk?

Reply:
answering myself - used in tests (requirements.txt is more like for dev dependencies)

Reply:
Looks like its used in the base integ class to tear down containers and images at the end of the test class runs. It also checks if the image exists already, and if not builds it: https://github.com/aws/aws-sam-build-images/blob/f47ef71c51208e409cc71418744258ebe077f42f/tests/build_image_base_test.py